src/settings.ts

```diff
@@ -304,6 +304,17 @@ export class ObsidianGitSettingsTab extends PluginSettingTab {
                     plugin.gitManager.updateGitPath(value || "git");
                 });
             });
+
+        new Setting(containerEl)
+            .setName("Base Path")
+            .addText((cb) => {
+                cb.setValue(plugin.settings.basePath);
+                cb.onChange((value) => {
+                    plugin.settings.basePath = value;
+                    plugin.saveSettings();
+                    plugin.gitManager.updateBasePath(value || "");
```

Comment:
I would first apply the new path and then save the setting. So when updating the base path fails, the setting doesn't get saved.

Reply:
I think the settings need to be saved first. The `updateBasePath` function doesn't actually use the arguments provided. This follows the same path as the Git Path.

https://github.com/mbseid/obsidian-git/blob/basePath/src/simpleGit.ts#L314-L320

Reply:
Ah good catch. Then keep your changes.